src/core/server/saved_objects/service/lib/filter_utils.ts

```diff
@@ -205,7 +205,23 @@ export const hasFilterKeyError = (
   return null;
 };
 
-const fieldDefined = (indexMappings: IndexMapping, key: string) => {
+export const fieldDefined = (indexMappings: IndexMapping, key: string) => {
   const mappingKey = 'properties.' + key.split('.').join('.properties.');
+  const potentialKey = get(indexMappings, mappingKey);
+
+  // If the `mappingKey` does not match a valid path, before returning null,
+  // we want to check and see if the intended path was for a multi-field
+  // such as `x.attributes.field.text` where `field` is mapped to both text
+  // and keyword
+  if (potentialKey == null) {
+    const indexOfLastProperties = mappingKey.lastIndexOf('properties');
+    const fieldMapping = mappingKey.substr(0, indexOfLastProperties);
+    // The 11 here refers to the length of `properties.`
+    const fieldType = mappingKey.substr(mappingKey.lastIndexOf('properties') + 11);
+
+    const mapping = `${fieldMapping}fields.${fieldType}`;
+    return get(indexMappings, mapping) != null;
+  }
+
   return get(indexMappings, mappingKey) != null;
```

Comment:
As the `if (potentialKey == null)` block already performs the assert, you can change the last line from 

`return get(indexMappings, mappingKey) != null;` 

to just 

`return true`